### PR TITLE
feat(LINGUIST-334): Added search user endpoint, updated user field username as unique

### PR DESCRIPTION
**Pull request by elif-krvan**
Initially, I put the search endpoint in the user controller. However, this was awkward since the endpoint URL was auth/search, and the endpoint had nothing to do with authentication. Therefore, I moved it to the profile controller.

### CODE REVIEW

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java [hunk 1]
```diff
@@ -39,4 +44,10 @@ public ResponseEntity<Object> getProfile(@RequestHeader(Header.USER_EMAIL) Strin
         return Response.create("OK", HttpStatus.OK, userProfile);        
     }
 
+    @GetMapping(consumes = MediaType.APPLICATION_JSON_VALUE, path = "search")
+    public ResponseEntity<Object> searchUser(@Valid @RequestBody QUserSearch userSearch,
```
Comment (KardelenCeren) [inline src/main/java/app/linguistai/bmvp/controller/ProfileController.java:48]: Please add swagger annotation for controllers

File: src/main/java/app/linguistai/bmvp/controller/AccountController.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
 import app.linguistai.bmvp.request.QResetPasswordSave;
 import app.linguistai.bmvp.service.gamification.UserStreakService;
 import app.linguistai.bmvp.service.stats.UserLoggedDateService;
+
+import org.springframework.data.domain.Page;
```
Comment (KardelenCeren) [inline src/main/java/app/linguistai/bmvp/controller/AccountController.java:14]: This seems to be an unused import

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java (hunk 1 above)
Comment (KardelenCeren) [inline src/main/java/app/linguistai/bmvp/controller/ProfileController.java:48]: @selimcanglr Currently search endpoint's parameters (username, page, size) need to be given in the request body. Would you prefer if they were given as part of the URL?

**Review by KardelenCeren — CHANGES_REQUESTED** [review]
Looks good, just left some minor comments

**Review by can-aslan — APPROVED** [review]
LGTM after addressing @KardelenCeren's comments

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java (hunk 1 above)
Comment (selimcanglr) [reply src/main/java/app/linguistai/bmvp/controller/ProfileController.java:48]: That would be more preferable, it doesn't matter that much.

File: src/main/java/app/linguistai/bmvp/controller/AccountController.java (hunk 2 above)
Comment (selimcanglr) [reply src/main/java/app/linguistai/bmvp/controller/AccountController.java:14]: @elif-krvan will look at it later

File: src/main/java/app/linguistai/bmvp/controller/ProfileController.java (hunk 1 above)
Comment (selimcanglr) [reply src/main/java/app/linguistai/bmvp/controller/ProfileController.java:48]: @elif-krvan will look at it later, I am merging this.

**Review by selimcanglr — APPROVED** [review]

**Pull request merged**